rest/docs/design/technical_design.md

```diff
@@ -0,0 +1,127 @@
+The `ActivateHandler`, uses the physical bytes generated by the `ProjectGenerator` to trigger a **build** invocation at **Openshift** level.  
+This means that we leverage **Openshift** "source to image (s2i)" capability, that accepts a reference to a source code project location, to build the project at source code level and produce also a corresponding **Docker** image including everything needed to spawn a runtime instance of our `Integration`.
+
+##### Deploy
+**Deploying**, as another responsibility of `ActivateHandler`. After a source code project corresponding to an `Integration` has been generated, built and shared to the interal repository, it still needs to be deployed to run and be available.
+This phase does just that: instructs **Openshift** platform to deploy a live incarnation of our logical `Integration`
+
+
+##### Deployment Data
+This phase actually occur before the `build` one, but it's better understood after what happens in the building phase has been made clearer.  
+We have seen that the building phase is all about code generation, to produce an output that will then pass further to a step of code compilation.  
+Before the code generation phase, actually some further manipulation happens.  
+The idea is to extract externalize from the code itself references to sensible information, like passwords, and externalized properties.  
+Secrets and properties will be turned into **Openshift** level handled resources, allowing them to follow a different lifecycle and letting a platform administrator  manage them.  
+These resources will be seen by the application as plain files; but in reality they are a logical resource, that **Openshift** re-exposes as a physical file for easier access.
+
+
+### Generated project details
+The project generated in the `build` phase is an **Apache Camel SpringBoot** based project, packed following an Apache Maven structure.  
+The Camel DSL used to define Camel Routes is a custom one, based initially on the `Funktion` project.  
+It uses a `.yaml` syntax.  
+A benefit from using this custom `.yaml` syntax, compared to the already present XML DSL is that it's easier to extend, from a development point of view. For example if you add a new `Step` kind.
+
+An important concept here is that the `.yaml` generated from the `Integration` might show a tree shape **similar but not necessarily identical** to the `.json` file that had been produced from the **Integration** conceptual model.  
+This is because some functionality, like for example **Technical Extensions** might need to create multiple **Steps** from a single definition, thus leading to a structure with a slightly different topology.
+
+If you are familiar with Camel framework you can find a correspondence between the following entitites:
+
+Flow --> Camel Route Builder
+Step --> Camel Route
```

Comment:
I thought that a Step can be different kind of things, not always a Route.
What about filters/data mapper?